### ISSUE TITLE
Move the Api Tokens to local storage instead of storing them in data.json

### ISSUE THREAD

**Issue by babeal** [logancyang/obsidian-copilot#902]
**Is your feature request related to a problem? Please describe.**

While encrypting the tokens is a step in the right direction, I still don't want credentials or derivatives of credentials stored on disk in the same file as the plugins general settings to be synced to cloud services.  

**Describe the solution you'd like**

Take a look at https://github.com/gavvvr/obsidian-imgur-plugin/blob/main/src/uploader/imgUploaderFactory.ts for a reasonable solution. The obsidian imgur plugin uses local storage to store the key for imagur after authentication and would be a good place for the api tokens. This way the key is not on disk, doesn't get backed up to obsidian data service, git, or icloud, and allows the sharing of the rest of the configuration.

**Describe alternatives you've considered**

Adding the data.json to .gitignore but then I have to reconfigure this plugin for every client I have.

**Additional context**

**Comment by logancyang** [comment]
Thanks for reporting this, and sorry for the slow reply.

Copilot v3 is in code freeze, so we're closing this as not planned. v4 is a ground-up rewrite and is launching soon, with a very different architecture around agent mode.

If this is still an issue for you in v4 once it lands, please open a new issue and we'll take a fresh look. 🙏
